packages/zoe/src/contracts/multipoolAutoswap/priceAuthority.js

```diff
@@ -0,0 +1,78 @@
+// @ts-check
+
+import { amountMath } from '@agoric/ertp';
+import { E } from '@agoric/eventual-send';
+import { Far } from '@agoric/marshal';
+import { observeNotifier } from '@agoric/notifier';
+import { assert, details as X } from '@agoric/assert';
+
+import { makeOnewayPriceAuthorityKit } from '../../contractSupport';
+
+export const makePriceAuthority = (
+  getOutputForGivenInput,
+  actualBrandIn,
+  actualBrandOut,
+  timer,
+  zcf,
+  notifier,
+  quoteIssuerKit,
+) => {
+  const { brand, issuer: quoteIssuer } = quoteIssuerKit;
+
+  /** @param {PriceQuoteValue} quote */
+  const authenticateQuote = quote => {
+    const quoteAmount = amountMath.make(quote, brand);
+    const quotePayment = quoteIssuerKit.mint.mintPayment(quoteAmount);
+    return harden({ quoteAmount, quotePayment });
+  };
+
+  // called by createQuote for each trigger to compare price to trigger level
+  const calcAmountOut = amountIn =>
+    getOutputForGivenInput(amountIn, actualBrandOut).amountOut;
+  const calcAmountIn = amountOut =>
+    getOutputForGivenInput(amountOut, actualBrandIn).amountIn;
```

Comment:
Again, should both of these `getOutputForGivenInput` calls use the same function?

Reply:
I think so, but the fact that calcAmountIn isn't actually used (`priceAuthority.js` lines 106 and 169) may be leaving me confused about the intended semantics.

Reply:
`calcAmountIn` is used: I think you missed `priceAuthority.js` line 257.

Reply:
I think the `calcAmountIn` is incorrect.  It should not calculate output, it should calculate input.  I'm fixing it the next PR.